libclc/generic/lib/atomic/atom_cmpxchg.cl

```diff
@@ -6,7 +6,28 @@
 //
 //===----------------------------------------------------------------------===//
 
-#include <clc/clc.h>
+#include <clc/atomic/atom_cmpxchg.h>
+#include <clc/atomic/atomic_cmpxchg.h>
+
+// cl_khr_global_int32_base_atomics
+#define IMPL(TYPE) \
+_CLC_OVERLOAD _CLC_DEF TYPE atom_cmpxchg(volatile global TYPE *p, TYPE cmp, TYPE val) { \
```

Comment:
Could we maybe hide `global` and `local` behind `IMPL(TYPE, ADDRSPACE)` and then have `IMPL(int, global)`, etc.?

Then we wouldn't need to `#undef` and re-`#def` `IMPL`.

Reply:
thanks, done in https://github.com/llvm/llvm-project/pull/134489/commits/3c3f06516b07c34b3211412904df05eef363fb40